File: math/1154_day_of_the_year/1154_day_of_the_year.go
package main

import (
	"fmt"
	"strconv"
	"strings"
)

func dayOfYear(date string) int {
	order := 0
	ds := strings.Split(date, "-")
	if len(ds) != 3 {
		return order
	}
	days := [12]int{31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31}
	y, _ := strconv.Atoi(ds[0])
	m, _ := strconv.Atoi(ds[1])
	d, _ := strconv.Atoi(ds[2])
	order += d
	for i := 0; i < m-1; i++ {
		order += days[i]
	}
	if m > 2 && (y%100 == 0 && y%400 == 0) || (y%100 != 0 && y%4 == 0) {
		order++
	}
	return order
}

func main() {
	res := dayOfYear("2019-01-09")
	fmt.Println(res)
}

File: math/1154_day_of_the_year/1154_day_of_the_year.py

class Solution(object):
    def dayOfYear(self, date):
        y, m, d = map(int, date.split('-'))
        days = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]
        if (y % 400) == 0 or ((y % 4 ==0 ) and (y % 100 != 0)):
            days[1] = 29
        return d + sum(days[:m-1])


date = "2019-01-09"
res = Solution().dayOfYear(date)
print(res)